carma-messenger-core/cpp_message/src/MobilityOperation_Message.cpp

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * CPP File containing Mobility Operation Message method implementations
+ */
+
+#include "MobilityOperation_Message.h"
+
+namespace Message_cpp
+{
+    boost::optional<cav_msgs::MobilityOperation> Mobility_Operation::decode_mobility_operation_message(std::vector<uint8_t>& binary_array){
+        
+        cav_msgs::MobilityHeader header;
+        cav_msgs::MobilityOperation output;
+        //decode results - stored in binary_array
+        asn_dec_rval_t rval;
+        MessageFrame_t* message=0;
```

Comment:
For clarity's sake, I'd recommend using `nullptr` here instead of the numerical `0`